pilot/pkg/config/kube/gateway/conditions.go

```diff
@@ -39,59 +39,79 @@ func createRouteStatus(gateways []routeParentReference, obj config.Config, curre
 	}
 	// Collect all of our unique parent references. There may be multiple when we have a route without section name,
 	// but reference a parent with multiple sections.
+	// While we process these internally for-each sectionName, in the status we are just supposed to report one merged entry
 	seen := map[k8s.ParentReference]routeParentReference{}
 	failedCount := map[k8s.ParentReference]int{}
-	for _, gw := range gateways {
+	successCount := map[k8s.ParentReference]int{}
+	for _, incoming := range gateways {
 		// We will append it if it is our first occurrence, or the existing one has an error. This means
 		// if *any* section has no errors, we will declare Admitted
-		if gw.DeniedReason != nil {
-			failedCount[gw.OriginalReference]++
+		if incoming.DeniedReason != nil {
+			failedCount[incoming.OriginalReference]++
+		} else {
+			successCount[incoming.OriginalReference]++
+		}
+		exist, f := seen[incoming.OriginalReference]
+		if !f {
+			exist = incoming
+		} else {
+			if incoming.DeniedReason == nil {
+				// The incoming gateway has no error; wipe out any errors
+				// When we have multiple, this means we attached without sectionName - we only need one success to be valid.
+				exist.DeniedReason = nil
+			} else if exist.DeniedReason != nil {
+				// TODO this only handles message, not reason
+				exist.DeniedReason.Message += "; " + incoming.DeniedReason.Message
+			}
+		}
+		seen[exist.OriginalReference] = exist
+	}
+	// Enhance error message when we have multiple
+	for k, gw := range seen {
+		if gw.DeniedReason == nil {
+			continue
 		}
-		if exist, f := seen[gw.OriginalReference]; !f || (exist.DeniedReason != nil && gw.DeniedReason == nil) {
-			seen[gw.OriginalReference] = gw
+		if failedCount[k] > 1 {
+			gw.DeniedReason.Message = fmt.Sprintf("failed to bind to %d parents, errors: %v", failedCount[k], gw.DeniedReason.Message)
```

Comment:
Almost all Message's here start with a Capital letter, even those for ConfigError's. 
Is this here desired?